### PR TITLE
Update to visual studio IDE project files

### PR DESCRIPTION
**Pull request by ghost**
I thought it would be a good idea to start updating the Visual Studio IDE files to the latest version. I had to make a tiny change to the network library loading and a tiny fix to PNG loading on the rom select screen. Could someone check to make sure these changes don't screw up the command line building?

### CODE REVIEW

File: src/dep/vs2010/pre.bat [hunk 1]
```diff
@@ -1,44 +1,12 @@
...
+perl license2rtf.pl -o ../../../projectfiles/visualstudio-2010/generated/license.rtf ../../license.txt
+perl cave_tile_func.pl -o ../../../projectfiles/visualstudio-2010/generated/cave_tile_func.h
+perl neo_sprite_func.pl -o ../../../projectfiles/visualstudio-2010/generated/neo_sprite_func.h
+perl psikyo_tile_func.pl -o ../../../projectfiles/visualstudio-2010/generated/psikyo_tile_func.h
+perl toa_gp9001_func.pl -o ../../../projectfiles/visualstudio-2010/generated/toa_gp9001_func.h
+perl gamelist.pl -o ../../../projectfiles/visualstudio-2010/generated/driverlist.h -l ../../../projectfiles/visualstudio-2010/generated/gamelist.txt ../../burn/drv/atari ../../burn/drv/capcom ../../burn/drv/cave ../../burn/drv/coleco ../../burn/drv/cps3 ../../burn/drv/dataeast ../../burn/drv/galaxian ../../burn/drv/irem ../../burn/drv/konami ../../burn/drv/megadrive ../../burn/drv/midway ../../burn/drv/msx ../../burn/drv/neogeo ../../burn/drv/pce ../../burn/drv/pgm ../../burn/drv/pst90s ../../burn/drv/pre90s ../../burn/drv/psikyo ../../burn/drv/sega ../../burn/drv/sg1000 ../../burn/drv/spectrum ../../burn/drv/taito ../../burn/drv/toaplan ../../burn/drv/snes ../../burn/drv/sms
```
Comment (dinkc64) [inline src/dep/vs2010/pre.bat:12]: just curious: are those hardcoded paths going to cause a problem for anyone?  Barry?

File: projectfiles/visualstudio-2010/fba_vs2010.vcxproj [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
-<Project DefaultTargets="Build" ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+<Project DefaultTargets="Build" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <ItemGroup Label="ProjectConfigurations">
```
Comment (dinkc64) [inline projectfiles/visualstudio-2010/fba_vs2010.vcxproj:3]: Does this change retain compatibility with the older MSVC?  (re: ToolsVersion="15.0")

File: src/dep/vs2010/pre.bat (hunk 1 above)
Comment (ghost) [reply src/dep/vs2010/pre.bat:12]: Barry may say differently but the paths are all relative from the .bat file so should be ok. I purposely .gitingore on the "generated" folder so it doesn't poo old headers in the repository. 

Also, I doubt he's using this .bat file as it seems to only be used in the IDE project files, but again I could be wrong.

File: projectfiles/visualstudio-2010/fba_vs2010.vcxproj (hunk 2 above)
Comment (ghost) [reply projectfiles/visualstudio-2010/fba_vs2010.vcxproj:3]: It should do, I tested it with vs2017 and vs2015 tfs as well as the latest sharpdevelop and it worked ok. However, the caveat there is that I have the latest Windows platform sdk installed which might be what this is referring too. Easily changed if someone is using older VS or wants to use the older tools (this should be the only change to the project file that would stop it working)

File: src/burn/snd/ymf262.cpp [hunk 3]
```diff
@@ -53,6 +53,10 @@ differences between OPL2 and OPL3 shown in datasheets:
 
 */
 
+#if defined (_MSC_VER)
```
Comment (ghost) [inline src/burn/snd/ymf262.cpp:56]: @dinkc64  I changed this to match the top of YM2612.c as M_PI is undefined in VS otherwise.

**Pull request merged**